### PR TITLE
Fix cancel button for create and edit schedule wizard

### PR DESCRIPTION
**Pull request by matoval**
Jira: https://issues.redhat.com/browse/AAP-22897

Fix cancel button for create and edit schedule wizard

### CODE REVIEW

File: frontend/awx/views/schedules/wizard/ScheduleEditWizard.tsx [hunk 1]
```diff
@@ -49,7 +49,12 @@ export function ScheduleEditWizard() {
     await Promise.resolve();
   };
 
-  const onCancel = () => navigate(location.pathname.replace('edit', 'details'));
+  const onCancel = () =>
+    navigate(
+      params?.id
+        ? location.pathname.replace('edit', 'details')
+        : `/templates/job_template/${schedule?.unified_job_template}/schedules/${schedule?.id}/details`
```
Comment (nixocio) [inline frontend/awx/views/schedules/wizard/ScheduleEditWizard.tsx:56]: Do we need to cover workflow_job_template as well?

Comment (matoval) [reply frontend/awx/views/schedules/wizard/ScheduleEditWizard.tsx:56]: Good catch! I'll update it.
Is job template and workflow job template the only two resource types when need to accommodate for the cancel button?

Comment (nixocio) [reply frontend/awx/views/schedules/wizard/ScheduleEditWizard.tsx:56]: @AlexSCorey Do you know this one?

**Review by akus062381 — APPROVED** [review]
Looks good!

File: frontend/awx/views/schedules/hooks/scheduleHelpers.tsx [hunk 2]
```diff
@@ -40,6 +41,84 @@ export const scheduleResourceTypeOptions: string[] = [
...
+  return useCallback(
+    (route: string, schedule: Schedule) => {
+      const unified_job_type = schedule.summary_fields.unified_job_template.unified_job_type;
+      const params = {
+        id: schedule.summary_fields.unified_job_template?.id.toString(),
+        schedule_id: schedule.id.toString(),
+      };
+
+      if (unified_job_type === 'inventory_update') {
+        const inventory_update_params = {
```
Comment (AlexSCorey) [inline frontend/awx/views/schedules/hooks/scheduleHelpers.tsx:55]: generally in JS/TS land we use camel case for variable names
```suggestion
        const inventoryUpdateParams = {
```

File: frontend/awx/views/schedules/hooks/scheduleHelpers.tsx [hunk 3]
```diff
@@ -40,6 +41,84 @@ export const scheduleResourceTypeOptions: string[] = [
...
+          name: t('Playbook run'),
+          details: { pageId: AwxRoute.JobTemplateScheduleDetails, params },
+          create: { pageId: AwxRoute.JobTemplateScheduleCreate, params },
+          edit: { pageId: AwxRoute.JobTemplateScheduleEdit, params },
+          resource: { pageId: AwxRoute.JobTemplateDetails, params },
+        }[route] as string | schedulePageUrl;
+      }
+      if (unified_job_type === 'project_update') {
+        return {
+          name: t('Source control update'),
```
Comment (AlexSCorey) [inline frontend/awx/views/schedules/hooks/scheduleHelpers.tsx:91]:
```suggestion
          name: t('Project update'),
```

File: frontend/awx/views/schedules/hooks/scheduleHelpers.tsx [hunk 4]
```diff
@@ -40,6 +41,84 @@ export const scheduleResourceTypeOptions: string[] = [
   'projects',
 ];
 
+export function useGetScheduleUrl() {
```
Comment (AlexSCorey) [inline frontend/awx/views/schedules/hooks/scheduleHelpers.tsx:44]: I think this function is large enough to move into its own file in the /hooks.

File: frontend/awx/views/schedules/hooks/scheduleHelpers.tsx [hunk 5]
```diff
@@ -40,6 +41,84 @@ export const scheduleResourceTypeOptions: string[] = [
...
+        }[route] as string | schedulePageUrl;
+      }
+      if (unified_job_type === 'workflow_job') {
+        return {
+          name: t('Workflow job'),
+          details: { pageId: AwxRoute.WorkflowJobTemplateScheduleDetails, params },
+          create: { pageId: AwxRoute.WorkflowJobTemplateScheduleCreate, params },
+          edit: { pageId: AwxRoute.WorkflowJobTemplateScheduleEdit, params },
+          resource: { pageId: AwxRoute.WorkflowJobTemplateDetails, params },
+        }[route] as string | schedulePageUrl;
```
Comment (AlexSCorey) [inline frontend/awx/views/schedules/hooks/scheduleHelpers.tsx:114]: I'm a fan of this approach!

**Review by AlexSCorey — CHANGES_REQUESTED** [review]
I really like your approach here.  Its good work.  

I've pulled down the code, that found a few bugs, and made a few suggestions on the code.  We should add some e2e testing to make sure the routing doesn't break in the future.  

When clicking on edit button in top level schedule list the url is missing a param.
Screenshot 2024-04-18 at 9 59 36 AM

Inside an inventory source schedule....on the details view.  When I click the tab `Back to schedules`
Screenshot 2024-04-18 at 10 01 16 AM

File: frontend/awx/views/schedules/wizard/ScheduleEditWizard.tsx [hunk 6]
```diff
@@ -49,7 +49,12 @@ export function ScheduleEditWizard() {
     await Promise.resolve();
   };
 
-  const onCancel = () => navigate(location.pathname.replace('edit', 'details'));
+  const onCancel = () =>
+    navigate(
+      params?.id
+        ? location.pathname.replace('edit', 'details')
```
[the remark this answers is not in the document]
Comment (matoval) [reply frontend/awx/views/schedules/wizard/ScheduleEditWizard.tsx:55]: I'll fix that, that was left over from the old approach 😬

**Review by AlexSCorey — APPROVED** [review]
This looks good to me.  Well done!

**Pull request merged**